tss-esapi/src/context.rs

```diff
@@ -431,6 +431,12 @@ impl Context {
                 Error::local_error(ErrorKind::MissingAuthSession)
             })
     }
+
+    /// Private function for handling that has been allocated with
+    /// C memory allocation functions in TSS.
+    fn ffi_data_to_owned<T>(data_ptr: *mut T) -> T {
+        MBox::into_inner(unsafe { MBox::from_raw(data_ptr) })
+    }
```

Comment:
This would've been ok as a macro as well, seems a bit weird to have it on `Context` given that it has nothing to do with `Context`, but feel free to leave as is.

Reply:
Not sure right now I got the result when I ran valgrind. One I remember was in the duplicate context method.

Reply:
Well it is only the context methods that currently have to deal with turning ffi data into owned data. So it is not too far fetched that it exists as a Context private associated function.